onnxruntime/core/providers/cpu/ml/treeregressor.h

```diff
@@ -6,14 +6,14 @@
 
 namespace onnxruntime {
 namespace ml {
-template <typename T>
+template <typename T, typename TH=T, typename TO=float>
```

Comment:
><typename T, typename TH=T, typename TO=float> [](http://example.com/codeflow?start=9&length=46)

would it be possible to divorce type depended code from independent? Then we could have smaller binary size and one kernel declaration

Reply:
It is not that easy. I wrote this implementation knowing I would have to do that change. 

Before, you had the following combinations for the regression: <float, float, float>, <double, float, float>. Then, 1 was added: <double, double, float>.

And for classifiers, before there were <float, float, float>, <double, float, float>, <int32, float, float>, <int64, float, float>. Added: <double, double, float>.

We can remove <int32, float, float> and replace it by <int64,float,float>.

Reply:
Would it be possible to use TypeList? This way the types supported are configurable and we can tune the binary size? Is ML included into minimal or mobile builds?

Reply:
It is part of the minimum build unless flag --disable_ml_pos is added to the build command line.